src/layer/vector/Polyline.js

```diff
@@ -164,12 +170,18 @@ export var Polyline = Path.extend({
 	// Adds a given point to the polyline. By default, adds to the first ring of
 	// the polyline in case of a multi-polyline, but can be overridden by passing
 	// a specific ring as a LatLng array (that you can earlier access with [`getLatLngs`](#polyline-getlatlngs)).
-	addLatLng: function (latlng, latlngs) {
-		latlngs = latlngs || this._defaultShape();
+	addLatLng: function (latlng, ring) {
+		var oldLatLngs = LineUtil.cloneLatLngs(this._latlngs);
```

Comment:
Cloning the whole array on every `addLatLng` is potentially a significant performance problem. Basically, any time you build out a route point by point, there will be extraordinary amount of redundant memory allocations and GC churn. 

I'm not sure what's the best approach here but maybe it's just not worth including `oldLatLngs` in the event in principle, and avoid any cloning.

Reply:
Is ok for me, we can add it later if it is needed. I will update the PR today